src/client/src/rt-components/Environment/EnvironmentContext.tsx

```diff
@@ -1,19 +1,39 @@
 import React from 'react'
 
-export interface EnvironmentValue {
-  isDesktop: Boolean
-  openfin?: Window
+export interface EnvironmentValue<Provider extends Window = Window> {
+  provider: Provider
```

Comment:
Instead of passing a Provider, create a Provider type that has the properties you're using (platform, type, etc).

Reply:
Unfortunately that breaks the typing — as this is preliminary its best to leave that to the subsequent improvements you'll bring.